### ISSUE TITLE
Receive remote signals?

### ISSUE THREAD

**Issue by LinqLover** [Skinah/EspMilightHub#9]
Hi, I'm an absolute newbie in terms of openHAB, so please forgive me any stupid questions :)

Can I use this binding to receive signals of a Milight Remote? I bought this model: [MiLight FUT089](https://greatecno.com/gb/illumination/104574-synergy-21-led-fernbedienung-rgb-ww-rgb-cct-8-zonen-milight-4038816069733.html)
![image](https://greatecno.com/119770-large_default/synergy-21-led-fernbedienung-rgb-ww-rgb-cct-8-zonen-milight.jpg)

I would like to control devices from other manufacturers with this remote. Is this possible at all, and can this binding help me to do this?

Many thanks in advance!

**Comment by Xento** [comment]
This should be possible.
You have to create items for the groups of the remote and than add items for the bulbs you want to control for example from HUE ...
Than in Openhab you can set this channels to follow the ones from the remote.

**Comment by LinqLover** [comment]
Thank you for your help! I'm really new at smart home, but this sounds feasible!

**Comment by Skinah** [comment]
there are two ways to get it running. One is to use the mqtt method found in this link and to not use the binding. The second way is to use the binding and after  it is decided to forward the events to another binding of your choice. There are people doing this so best to ask in the forums and it may take a while for someone to reply. You can bind two channels from different bindings from memory to an item but I am no expert in doing the advanced stuff in openhab.

https://community.openhab.org/t/in-wall-rgbw-dimmer-that-sends-a-signal-to-openhab/41641/7

**Comment by LinqLover** [comment]
So for a simple start, which option would you recommend? :)

**Comment by Skinah** [comment]
I have never done it so can’t confirm it works how I think it does but by linking two bindings it should move a control from one binding and the output of the control feeds another. I would try this first as it will be easier and faster to setup. Ask about it on the forum after you read the documentation as I have never done it.

The other method will work and I have done it it
Just requires you to use rules  which plenty of
Examples are in the thread.

**Comment by LinqLover** [comment]
Hi @Skinah, maybe you can help me again? Meanwhile, I moved my setup to openHABian (raspi) and installed mosquitto via openhabian-config. I copied both jar files from http://www.pcmus.com/openhab/EspMilightHubBinding/ into the add-ons folder.

Following problem:  
My log shows an authentication error (MqttSecurityException) from EspMilightHubBridgeHandler every 10 seconds, stating that I am not authorized to connect:

```
20:13:00.781 [ERROR] [ub.handler.EspMilightHubBridgeHandler] - Error: Could not connect to MQTT broker.{}
org.eclipse.paho.client.mqttv3.MqttSecurityException: Not authorized to connect
        at org.eclipse.paho.client.mqttv3.internal.ExceptionHelper.createMqttException(ExceptionHelper.java:28) ~[?:?]
        at org.eclipse.paho.client.mqttv3.internal.ClientState.notifyReceivedAck(ClientState.java:1040) ~[?:?]
        at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:151) ~[?:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]
```

Where do I have to which connection options? I tried the following variants:
- stored  mosquitto password for the user openhabian in the `/etc/openhab2/services/mqtt.cfg` file
- stored  password for the milight hub in the `/etc/openhab2/services/mqtt.cfg` file
- no mqtt.cfg at all.

All failing.

Futhermore, on http://milight-hub/, I can watch the signals from my remote when sniffing is activated.
![image](https://user-images.githubusercontent.com/38782922/73137673-0814bd80-405b-11ea-9827-a2d5c3bf894b.png)
However, running `mosquitto_sub -u openhabian -P mymosquittopassword -p 1883 -v -t 'milight/#'` does not show any events when I press a key on the remote. Is my assumption correct that this cannot work until the authentication problem is solved?

Would be great if you could give me some hints! :-)

**Comment by Skinah** [comment]
Please post on a forum so others can reply. This is not a binding issue.

**Comment by LinqLover** [comment]
Sorry! My latest questions in the openHAB forum were not answered, but I can give another try ...

**Discussion moved [here](https://community.openhab.org/t/authenticate-paho-mqtt-client-for-espmilighthub-using-mosquitto/91439).**

**Comment by LinqLover** [comment]
For your or anyone's other information - in the meantime, I managed to set up a simple working solution. It's not nice, it rather quite ugly, but it works for me: https://community.openhab.org/t/authenticate-paho-mqtt-client-for-espmilighthub-using-mosquitto/91439/17?u=linqlover

By creating each one thing, item, and rule per group, I also managed to receive the signals of each remote's group. Just writing these lines because I hope to help someone else having the same problems anywhen later. :)
